### ISSUE TITLE
New installation instructions bug: Unknown function: fzf#install

### ISSUE THREAD

**Issue by mjlbach** [junegunn/fzf.vim#1008]
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [x] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

### Problem ###
When using the following minimal vimrc, fzf fails to install with the error:
```
x Post-update hook for fzf ... Vim(return):E117: Unknown function: fzf#install
```

### Minimal vimrc ###
```
set nocompatible hidden laststatus=2

if !filereadable('/tmp/plug.vim')
  silent !curl --insecure -fLo /tmp/plug.vim
    \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
endif

source /tmp/plug.vim
call plug#begin('/tmp/plugged')
Plug 'junegunn/fzf', { 'do': { -> fzf#install() } }
Plug 'junegunn/fzf.vim'
call plug#end()

autocmd VimEnter * PlugClean! | PlugUpdate --sync
```
### Workaround ###
The old style command
```
Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
```
seems to work fine still

**Comment by junegunn** [comment]
Interestingly, the error only occurs when vim-plug installer is automatically run during startup, which means, if you comment out `VimEnter` autocmd and invoke `PlugInstall` manually, it works as expected. To fix this, we'll have to change the way vim-plug works during Vim startup.

FYI, `fzf#install()` is equivalent to `./install --bin` (download binary only). The advantage of the former is that it's cross-platform.

**Comment by mjlbach** [comment]
Hmm, I noticed the error originally when invoking vim-plug manually from Neovim (not automatically during startup). I can try to reproduce/dig to see what's happening...

**Comment by junegunn** [comment]
It should be fixed now.

**Comment by horseinthesky** [comment]
It is broken again =(
```
E117: Unknown function: fzf#install
```

My config:
```
Plug 'junegunn/fzf', { 'do': { -> fzf#install() } }
Plug 'junegunn/fzf.vim'
```

Old way works:
```
Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
Plug 'junegunn/fzf.vim'
```

NVIM 0.4.3 in Ubuntu 20.04.

**Comment by cwfparsonson** [comment]
I am also now getting an 'Unknown function: fzf#install' error with:

```
Plug 'junegunn/fzf', { 'do': { -> fzf#install() } }
Plug 'junegunn/fzf.vim'
```

in my init.vim config. 

Many thanks to @horseinthesky for pointing to the solution, replacing the above with:

```
Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
Plug 'junegunn/fzf.vim'
```

works.

**Comment by junegunn** [comment]
> but does not install the binary and shell completion files

`fzf#install` is identical to `fzf --bin` not `fzf --all`. It only downloads the binary so that the Vim plugin works as expected, it doesn't do anything beyond that.

**Comment by horseinthesky** [comment]
> > but does not install the binary and shell completion files
> 
> `fzf#install` is identical to `./install --bin` not `./install --all`. It only downloads the binary so that the Vim plugin works as expected, it doesn't do anything beyond that.

Got it. Thank you.

**Comment by pdfrod** [comment]
Same problem here. Should this issue be re-opened or create a new one?

**Comment by elzii** [comment]
Make sure `:echo &shell` is actually pointing to something that exists! In my case this was from porting over a linux config to macos.

**Comment by Anogoya-Dagaate** [comment]
`:echo &shell` returns /usr/bin/bash; command :fzf opens a Windows command prompt for some reason. Been reading a tonne of attempts at resolving this issue on Windows running Vim from Git Bash i.e. launching Vim from inside of the bash shell instead of launching Vim from command prompt or powershell. Neither has resolved the issue. Hopefully find a fix soon

**Comment by sleepyhollo** [comment]
First of all junegunn, you are great! your plugins are life.

I am trying this on Windows 10 with Vim 8.2 (Included patches: 1-1883). I am trying to install this using VimPlug. I tried both `Plug 'junegunn/fzf', { 'do': { -> fzf#install() } }` and `Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --bin' }` and both failed. Unfortunately i did not save the error messages. 

I finally got it working by running the install.ps1 script manually from the fzf install folder.
